README.md

```diff
@@ -18,25 +20,184 @@ $ npm install --save pipage
 
 ## Usage
 
+For detailed API documentation, see [doc/](doc/)
+
+```js
+var Pipage = require('pipage')
+```
+
+The pipeline is essentially a duplex stream with internal streams, which can be **added to**, **removed from** and **spliced** at runtime:
+
+```
+readable -> pipage[ transform, transform, ... ] -> writable
+```
+
+### Creating a pipeline
+
+```js
+// A blank pass-through pipeline:
+var pipeline = new Pipage()
+
+// Initialized with already existing streams:
+var pipeline = new Pipage([ transform1, transform2, ... ])
+```
+
+### Adding streams
+
+```js
+// Streams can be appended,
+pipeline.append( stream )
+pipeline.append( stream1, stream2, ..., streamN )
+
+// prepended,
+pipeline.prepend( stream1, stream2, ..., streamN )
+
+// inserted at a specific index,
+pipeline.insert( 3, stream1, ..., streamN )
+
+// which is the same as splicing in streams (add N streams at index 3):
+pipeline.splice( 3, 0, stream1, ..., streamN )
+```
+
+### Removing streams
+
+```js
+// Streams can be shifted off the beginning,
+var firstStream = pipeline.shift()
+
+// or popped off the end,
+var lastStream = pipeline.pop()
+
+// spliced out at a specific index (remove 3 from index 2),
+var removedStreams = pipeline.splice( 2, 3 )
+
+// or removed by reference:
+pipeline.remove( stream )
+```
+
+### Selecting streams
+
+```js
+// Get a stream at a specific index in the pipeline:
+var stream = pipeline.get( 2 )
+var lastStream = pipeline.get( -1 )
+
+// Find a stream in the pipeline:
+var index = pipeline.indexOf( stream )
+var lastIndex = pipeline.lastIndexOf( stream )
+```
+
+### Example
+
+```js
+var Pipage = require('pipage')
+var path = require('path')
+var zlib = require('zlib')
+var unbzip2 = require('unbzip2-stream')
+var xz = require('xz')
+
+var pipeline = new Pipage()
+
+switch( path.extname( filename ) ) {
+  case '.gz':
+    pipeline.prepend( zlib.createUnzip() );
```

Comment:
It's using `.prepend()`, because that's the first thing you'd want to do before doing anything else with the data when decompressing something, if that makes sense.